extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaIndexTaskClient.java

```diff
@@ -143,147 +111,147 @@ public void resume(TaskLocation location, String id)
-          "http",
-          null,
-          location.getHost(),
-          location.getPort(),
-          String.format("%s/%s/%s", BASE_PATH, id, "offsets/current"),
-          null,
-          null
-      );
-      final StatusResponseHolder response = submitRequest(new Request(HttpMethod.GET, serviceUri.toURL()), retry);
-
+      final FullResponseHolder response = submitRequest(id, HttpMethod.GET, "offsets/current", null, retry);
       return jsonMapper.readValue(response.getContent(), new TypeReference<Map<Integer, Long>>() {});
     }
-    catch (Exception e) {
+    catch (IOException e) {
       throw Throwables.propagate(e);
     }
   }
 
-  public Map<Integer, Long> getEndOffsets(TaskLocation location, String id)
+  public Map<Integer, Long> getEndOffsets(String id)
   {
     try {
-      final URI serviceUri = new URI(
-          "http",
-          null,
-          location.getHost(),
-          location.getPort(),
-          String.format("%s/%s/%s", BASE_PATH, id, "offsets/end"),
-          null,
-          null
-      );
-      final StatusResponseHolder response = submitRequest(new Request(HttpMethod.GET, serviceUri.toURL()), true);
-
+      final FullResponseHolder response = submitRequest(id, HttpMethod.GET, "offsets/end", null, true);
       return jsonMapper.readValue(response.getContent(), new TypeReference<Map<Integer, Long>>() {});
     }
-    catch (Exception e) {
+    catch (IOException e) {
       throw Throwables.propagate(e);
     }
   }
 
-  public void setEndOffsets(TaskLocation location, String id, Map<Integer, Long> endOffsets)
+  public void setEndOffsets(String id, Map<Integer, Long> endOffsets)
   {
-    setEndOffsets(location, id, endOffsets, false);
+    setEndOffsets(id, endOffsets, false);
   }
 
-  public void setEndOffsets(TaskLocation location, String id, Map<Integer, Long> endOffsets, boolean resume)
+  public void setEndOffsets(String id, Map<Integer, Long> endOffsets, boolean resume)
   {
     try {
-      final URI serviceUri = new URI(
-          "http",
-          null,
-          location.getHost(),
-          location.getPort(),
-          String.format("%s/%s/%s", BASE_PATH, id, "offsets/end"),
-          resume ? "resume=true" : null,
-          null
-      );
       submitRequest(
-          new Request(HttpMethod.POST, serviceUri.toURL()).setContent(
-              MediaType.APPLICATION_JSON,
-              jsonMapper.writeValueAsBytes(endOffsets)
-          ), true
+          id,
+          HttpMethod.POST,
+          "offsets/end",
+          resume ? "resume=true" : null,
+          jsonMapper.writeValueAsBytes(endOffsets),
+          true
       );
     }
-    catch (Exception e) {
+    catch (IOException e) {
       throw Throwables.propagate(e);
     }
   }
 
-  private StatusResponseHolder submitRequest(Request request, boolean retry)
+  @VisibleForTesting
+  RetryPolicyFactory getRetryPolicyFactory() {
+    return new RetryPolicyFactory(new RetryPolicyConfig()
```

Comment:
does this need to create a new retrypolicyfactory each time?
Also, where do the defaults come from?

Perhaps this method should be called createRetryPolicyFactory()